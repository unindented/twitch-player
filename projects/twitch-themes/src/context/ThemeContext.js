import PropTypes from "prop-types";
import React, { createContext } from "react";
import { light } from "../themes";

export const ThemeContext = createContext(light);

export const ThemeProvider = ({ theme, ...props }) => (
  <ThemeContext.Provider {...props} value={theme || light} />
);

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};
