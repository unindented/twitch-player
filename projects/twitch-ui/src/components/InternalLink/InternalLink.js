import PropTypes from "prop-types";
import React, { useCallback } from "react";
import { useRouter } from "../../hooks";
import Link from "../Link";

const InternalLink = props => {
  const { history } = useRouter();

  const handlePress = useCallback(
    evt => {
      evt.preventDefault();
      history.push(props.href);
    },
    [props.href]
  );

  return <Link {...props} onPress={handlePress} />;
};

InternalLink.propTypes = {
  href: PropTypes.string.isRequired,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.arrayOf(PropTypes.object),
  ]),
  children: PropTypes.node.isRequired,
  testID: PropTypes.string,
};

export default InternalLink;
