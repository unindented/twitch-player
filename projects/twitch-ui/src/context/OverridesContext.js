import PropTypes from "prop-types";
import React, { createContext } from "react";
import ExternalLink from "../components/ExternalLink";

const defaultOverrides = { Link: ExternalLink };

export const OverridesContext = createContext(defaultOverrides);

export const OverridesProvider = ({ overrides, ...props }) => (
  <OverridesContext.Provider {...props} value={overrides || defaultOverrides} />
);

OverridesProvider.propTypes = {
  overrides: PropTypes.object,
};
