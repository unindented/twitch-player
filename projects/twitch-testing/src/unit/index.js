import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
