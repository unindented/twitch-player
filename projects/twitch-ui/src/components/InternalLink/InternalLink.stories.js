import { storiesOf } from "@storybook/react";
import React from "react";
import { MemoryRouter as Router } from "react-router";
import InternalLink from ".";

storiesOf("Components|InternalLink", module).add("default", () => (
  <Router>
    <InternalLink href="/settings">Settings</InternalLink>
  </Router>
));
