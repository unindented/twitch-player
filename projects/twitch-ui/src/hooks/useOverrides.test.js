import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import { MemoryRouter as Router } from "react-router";
import InternalLink from "../components/InternalLink";
import { OverridesProvider } from "../context";
import { useOverrides } from ".";

const SomeComponent = () => {
  const { Link } = useOverrides();

  return <Link href="/some-link">Some Link</Link>;
};

describe("useOverrides", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <OverridesProvider overrides={{ Link: InternalLink }}>
        <Router>
          <SomeComponent />
        </Router>
      </OverridesProvider>
    );
  });

  it("renders with the overriden link", () => {
    expect(instance.asFragment()).toMatchSnapshot();
  });
});
