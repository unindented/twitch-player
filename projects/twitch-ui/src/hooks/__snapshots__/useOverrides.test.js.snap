// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useOverrides renders with the overriden link 1`] = `
<DocumentFragment>
  <a
    class="rn-backgroundColor-1niwhzg rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-1471scf rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    data-focusable="true"
    data-testid="link"
    dir="auto"
    href="/some-link"
    role="link"
    style="color: rgb(100, 65, 164); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif;"
  >
    Some Link
  </a>
</DocumentFragment>
`;
